src/Emulator.ts

```diff
@@ -106,8 +110,20 @@ implements IGetGenesisInfos, IGetProtocolParameters, IResolveUTxOs, ISubmitTx
     //       - If no:
     //          - remove the transaction from the mempool
     // Moves time forward to the next blockNumber block
-    awaitBlock (blockNum : number) : void {
-        if (blockNum > 0){
+
+    // height in blockchain is the block number. so if awaitBlock(1) represents to wait for one block to go through 
+
+    awaitBlock (height : number) : void {
+        if (height > 0){
+
+            // const slotsToAdvance = blockNum - this.slot; // Calculate slots to advance
+            // if (slotsToAdvance <= 0) {
+            //     console.warn("Invalid block number. Must be greater than the current slot.");
+            //     return;
+            // }
+            this.slot += height * 20;
+            this.time += height * 20 * 1000;
+
             while (!this.mempool.isEmpty()){
```

Comment:
Ok, so this is where you should take into account the size of the block you're producing.
In pseudo code it should look something like this:
currentBlockUsed = 0
nextTxSize = getTxSize(mempool.head())
while (currentBlockUsed + nextTxsize < maxBlockSize){
  currentBlockUsed += nextTxsize
  ...
  nextTxSize = getTxSize(mempool.head())
}

Or something like that

Reply:
And then have a loop wrap around with like:
````
while height >0
   height -= 1
   while (....
````

Reply:
Updated for review